openmetadata-service/src/main/java/org/openmetadata/service/util/S3Util.java

```diff
@@ -0,0 +1,109 @@
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.core.ResponseInputStream;
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.GetObjectRequest;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+import software.amazon.awssdk.services.s3.model.S3Exception;
+
+public class S3Util {
+
+  S3Client client;
+
+  private String getOrDefault(String original, String defaultStr) {
+    return Strings.isNotEmpty(original) ? original : defaultStr;
+  }
+
+  private void initClient(SampleDataS3Config s3Config) {
+    if (s3Config != null
+        && s3Config.getAwsConfig() != null
+        && Strings.isNotEmpty(s3Config.getAwsConfig().getAwsRegion())) {
+      SecretsManagerFactory.getSecretsManager().decryptSampleDataS3Config(s3Config);
+      String region = s3Config.getAwsConfig().getAwsRegion();
+      String accessKeyId = this.getOrDefault(s3Config.getAwsConfig().getAwsAccessKeyId(), "");
+      String secretAccessKey = this.getOrDefault(s3Config.getAwsConfig().getAwsSecretAccessKey(), "");
+      AwsCredentialsProvider credentialsProvider;
+      if (Strings.isBlank(accessKeyId) && Strings.isBlank(secretAccessKey)) {
+        credentialsProvider = DefaultCredentialsProvider.create();
+      } else {
+        credentialsProvider =
+            StaticCredentialsProvider.create(AwsBasicCredentials.create(accessKeyId, secretAccessKey));
+      }
+      this.client = S3Client.builder().region(Region.of(region)).credentialsProvider(credentialsProvider).build();
+    } else {
+      this.client = S3Client.create();
+    }
+  }
+
+  private String getSampleDataObjectKey(SampleDataS3Config s3Config, Table table) {
+    return getOrDefault(s3Config.getPrefix(), "")
+        + "/"
+        + table.getService().getName()
+        + "/"
+        + table.getDatabase().getName()
+        + "/"
+        + table.getDatabaseSchema().getName()
+        + "/"
+        + table.getName()
+        + "/sample_data.json";
+  }
+
+  public void uploadSampleData(SampleDataS3Config s3Config, TableData tableData, Table table) {
+    this.initClient(s3Config);
+    try {
+      String objectKey = this.getSampleDataObjectKey(s3Config, table);
+      PutObjectRequest putOb = PutObjectRequest.builder().bucket(s3Config.getBucketName()).key(objectKey).build();
+      ObjectMapper objectMapper = new ObjectMapper();
+      byte[] bytesToWrite = objectMapper.writeValueAsBytes(tableData);
+      this.client.putObject(putOb, RequestBody.fromBytes(bytesToWrite));
+    } catch (JsonProcessingException e) {
+      throw new RuntimeException(e);
+    } catch (S3Exception e) {
+      System.err.println(e.getMessage());
+    }
+  }
+
+  public TableData getSampleData(SampleDataS3Config s3Config, Table table) {
+    this.initClient(s3Config);
```

Comment:
not sure if this is a bit dangerous. Imagine we have 2 profilers running at the same time, each pointing to a different bucket. We'd be overriding the internal client of the class.

One approach could be to define an internal `Map<S3Config, Client>` so that we pick up the right internal client in each case